lib/rules/no-duplicate-id.js

```diff
@@ -5,7 +5,7 @@ import Rule from './_base.js';
 class ConditionalScope {
   constructor() {
     this.seenIdStack = [new Set()];
-
+    this.ignoreIds = new Set(['{{unique-id}}']);
```

Comment:
Since this will never change during runtime and isn't related to the scope class, shall we just define it outside the class as `IGNORE_IDS` or `ID_ALLOWLIST`?